generator/rs/lib.tmpl.rs

```diff
@@ -1,7 +1,63 @@
 #![deny(clippy::all)]
 
-use napi::{Env, JsObject, Result};
-use napi_derive::module_exports;
+use napi::{Env, JsObject, JsUndefined, JsFunction, Result};
+use napi_derive::{module_exports, napi};
+
+use std::sync::mpsc::{self, Receiver, Sender};
+use std::sync::{Mutex, OnceLock};
+
+pub(crate) struct SendItem(*mut f32);
+unsafe impl Send for SendItem {}
+pub(crate) fn send_recv_pair() -> &'static Mutex<(Option<Sender<SendItem>>, Option<Receiver<SendItem>>)> {
+    static PAIR: OnceLock<Mutex<(Option<Sender<SendItem>>, Option<Receiver<SendItem>>)>> = OnceLock::new();
+    PAIR.get_or_init(|| {
+        let (send, recv) = mpsc::channel();
+        Mutex::new((Some(send), Some(recv)))
+    })
+}
+
+#[napi]
+pub fn run_audio_worklet(env: Env) -> Result<JsUndefined> {
+    println!("inside rust worklet");
+    let recv = send_recv_pair().lock().unwrap().1.take().unwrap();
+    for item in recv {
+        let proc = env.get_global()?.get_property::<_, JsObject>(env.create_string("proc123")?)?;
+        let process = proc.get_property::<_, JsFunction>(env.create_string("process")?)?;
+
+        let data: &mut[u8] = unsafe {
+            std::slice::from_raw_parts_mut(item.0 as *mut _, 128 * 4)
+        };
+        let data_ptr = data.as_mut_ptr();
+        let ptr_length = data.len();
+        let output_samples = unsafe {
+            env
+                .create_arraybuffer_with_borrowed_data(
+                    data_ptr,
+                    ptr_length,
+                    (),
+                    napi::noop_finalize,
+                )
+                .map(|array_buffer| {
+                    array_buffer
+                        .into_raw()
+                        .into_typedarray(napi::TypedArrayType::Float32, 128, 0)
+                })
+            .unwrap()
+        };
+
+        let mut output_channels = env.create_array(0)?;
+        output_channels.insert(output_samples)?;
+        let mut outputs = env.create_array(0)?;
+        outputs.insert(output_channels)?;
+
+        let _ret: bool = process.call3(
```

Comment:
I think we need apply3 to actually bind `this`

Reply:
I guess we will need monkey patching on the JS side to have the right context / this when `run_audio_worklet` is called

Reply:
My idea is that we keep a list of all worklet processors on the rust side, and use some extra fields in the message channel to select the proper processor for the render call.

Reply:
Makes sens, and your solution is the most straightforward indeed, from the documentation I was afraid we could only pass the original `Context` to `apply`.